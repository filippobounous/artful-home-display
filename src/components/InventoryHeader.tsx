import { SidebarTrigger } from '@/components/ui/sidebar';
import { DarkModeToggle } from '@/components/DarkModeToggle';
import { ApiHealthIndicator } from '@/components/ApiHealthIndicator';
import { useDashboardApiHealth } from '@/hooks/useDashboardApiHealth';
import { useSystemState } from '@/hooks/useSystemState';
import { cn } from '@/lib/utils';

export function InventoryHeader() {
  const { showApiHealth } = useDashboardApiHealth();
  const { topBarState } = useSystemState();

  const getTopBarClasses = () => {
    switch (topBarState) {
      case 'testing':
        return cn(
          'bg-[hsl(var(--tb-bg-testing)/0.15)] text-[hsl(var(--tb-fg-default))]',
          'dark:bg-[hsl(var(--tb-bg-testing)/0.2)] dark:text-[hsl(var(--tb-fg-contrast))]',
        );
      case 'apiwarn':
        return cn(
          'bg-[hsl(var(--tb-bg-apiwarn)/0.15)] text-[hsl(var(--tb-fg-default))]',
          'dark:bg-[hsl(var(--tb-bg-apiwarn)/0.2)] dark:text-[hsl(var(--tb-fg-contrast))]',
        );
      default:
        return cn(
          'bg-[hsl(var(--tb-bg-default)/0.95)] text-[hsl(var(--tb-fg-default))]',
        );
    }
  };

  return (
    <header
      className={cn(
        'border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar',
        'border-[hsl(var(--tb-border))]',
        getTopBarClasses(),
      )}
    >
      <div className="flex w-full items-center justify-between px-4">
        <div className="flex items-center gap-4">
          <SidebarTrigger />
          {showApiHealth && <ApiHealthIndicator enablePolling={true} />}
        </div>

        <div className="flex items-center gap-3">
          <DarkModeToggle />
        </div>
      </div>
    </header>
  );
}
