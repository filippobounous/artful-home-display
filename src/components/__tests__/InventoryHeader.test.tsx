import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { InventoryHeader } from '../InventoryHeader';

let mockState: 'default' | 'testing' | 'apiwarn' = 'default';

vi.mock('@/hooks/useSystemState', () => ({
  useSystemState: () => ({ topBarState: mockState }),
}));

vi.mock('@/hooks/useDashboardApiHealth', () => ({
  useDashboardApiHealth: () => ({ showApiHealth: false }),
}));

vi.mock('@/components/ui/sidebar', () => ({
  SidebarTrigger: () => <div />,
}));

vi.mock('@/components/DarkModeToggle', () => ({
  DarkModeToggle: () => <div />,
}));

vi.mock('@/components/ApiHealthIndicator', () => ({
  ApiHealthIndicator: () => <div />,
}));

describe('InventoryHeader visual states', () => {
  const renderHeader = (theme?: string) =>
    render(
      <div className={theme}>
        <InventoryHeader />
      </div>,
    );

  it('renders default light', () => {
    mockState = 'default';
    const { asFragment } = renderHeader();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders testing light', () => {
    mockState = 'testing';
    const { asFragment } = renderHeader();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders apiwarn light', () => {
    mockState = 'apiwarn';
    const { asFragment } = renderHeader();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders default dark', () => {
    mockState = 'default';
    const { asFragment } = renderHeader('dark');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders testing dark', () => {
    mockState = 'testing';
    const { asFragment } = renderHeader('dark');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders apiwarn dark', () => {
    mockState = 'apiwarn';
    const { asFragment } = renderHeader('dark');
    expect(asFragment()).toMatchSnapshot();
  });
});
