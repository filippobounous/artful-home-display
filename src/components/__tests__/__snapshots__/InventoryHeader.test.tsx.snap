// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InventoryHeader visual states > renders apiwarn dark 1`] = `
<DocumentFragment>
  <div
    class="dark"
  >
    <header
      class="border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar border-[hsl(var(--tb-border))] bg-[hsl(var(--tb-bg-apiwarn)/0.15)] text-[hsl(var(--tb-fg-default))] dark:bg-[hsl(var(--tb-bg-apiwarn)/0.2)] dark:text-[hsl(var(--tb-fg-contrast))]"
    >
      <div
        class="flex w-full items-center justify-between px-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <div />
        </div>
        <div
          class="flex items-center gap-3"
        >
          <div />
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`InventoryHeader visual states > renders apiwarn light 1`] = `
<DocumentFragment>
  <div>
    <header
      class="border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar border-[hsl(var(--tb-border))] bg-[hsl(var(--tb-bg-apiwarn)/0.15)] text-[hsl(var(--tb-fg-default))] dark:bg-[hsl(var(--tb-bg-apiwarn)/0.2)] dark:text-[hsl(var(--tb-fg-contrast))]"
    >
      <div
        class="flex w-full items-center justify-between px-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <div />
        </div>
        <div
          class="flex items-center gap-3"
        >
          <div />
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`InventoryHeader visual states > renders default dark 1`] = `
<DocumentFragment>
  <div
    class="dark"
  >
    <header
      class="border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar border-[hsl(var(--tb-border))] bg-[hsl(var(--tb-bg-default)/0.95)] text-[hsl(var(--tb-fg-default))]"
    >
      <div
        class="flex w-full items-center justify-between px-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <div />
        </div>
        <div
          class="flex items-center gap-3"
        >
          <div />
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`InventoryHeader visual states > renders default light 1`] = `
<DocumentFragment>
  <div>
    <header
      class="border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar border-[hsl(var(--tb-border))] bg-[hsl(var(--tb-bg-default)/0.95)] text-[hsl(var(--tb-fg-default))]"
    >
      <div
        class="flex w-full items-center justify-between px-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <div />
        </div>
        <div
          class="flex items-center gap-3"
        >
          <div />
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`InventoryHeader visual states > renders testing dark 1`] = `
<DocumentFragment>
  <div
    class="dark"
  >
    <header
      class="border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar border-[hsl(var(--tb-border))] bg-[hsl(var(--tb-bg-testing)/0.15)] text-[hsl(var(--tb-fg-default))] dark:bg-[hsl(var(--tb-bg-testing)/0.2)] dark:text-[hsl(var(--tb-fg-contrast))]"
    >
      <div
        class="flex w-full items-center justify-between px-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <div />
        </div>
        <div
          class="flex items-center gap-3"
        >
          <div />
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`InventoryHeader visual states > renders testing light 1`] = `
<DocumentFragment>
  <div>
    <header
      class="border-b backdrop-blur supports-[backdrop-filter]:bg-background/60 sticky top-0 z-40 topbar border-[hsl(var(--tb-border))] bg-[hsl(var(--tb-bg-testing)/0.15)] text-[hsl(var(--tb-fg-default))] dark:bg-[hsl(var(--tb-bg-testing)/0.2)] dark:text-[hsl(var(--tb-fg-contrast))]"
    >
      <div
        class="flex w-full items-center justify-between px-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <div />
        </div>
        <div
          class="flex items-center gap-3"
        >
          <div />
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
